### PR TITLE
JDK-8272944: Use snippets in jdk.javadoc documentation

### PR DESCRIPTION
**Pull request by jonathan-gibbons**
Please review a patch to convert an example in the `jdk.javadoc` documentation to use snippets instead of raw HTML and/or `{@code }`.  There are some minor tweaks to the text of the example, but no changes to the normative specification of the package.

Given that the example is in `package-info.java`, a minor change is required in `JavacElements` to ensure that the package symbol is completed (i.e. the `package-info.java` file has been read) when accessing the `PackageElement`.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8272944](https://bugs.openjdk.java.net/browse/JDK-8272944): Use snippets in jdk.javadoc documentation

### Reviewers
 * [Hannes Wallnöfer](https://openjdk.java.net/census#hannesw) (@hns - **Reviewer**) ⚠️ Review applies to f99df7fc8189002d1086628d9c58f55122cb27b1

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6685/head:pull/6685` \
`$ git checkout pull/6685`

Update a local copy of the PR: \
`$ git checkout pull/6685` \
`$ git pull https://git.openjdk.java.net/jdk pull/6685/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6685`

View PR using the GUI difftool: \
`$ git pr show -t 6685`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6685.diff">https://git.openjdk.java.net/jdk/pull/6685.diff</a>

</details>

### CODE REVIEW

File: src/jdk.javadoc/share/classes/jdk/javadoc/doclet/package-info.java [hunk 1]
```diff
@@ -51,9 +51,9 @@
  * The invocation is defined by the interface {@link jdk.javadoc.doclet.Doclet}
  * -- the {@link jdk.javadoc.doclet.Doclet#run(DocletEnvironment) run} interface
  * method, defines the entry point.
- * <pre>
- *    public boolean <b>run</b>(DocletEnvironment environment)
- * </pre>
+ * {@snippet id="entry-point" type=java :
```
Comment (FrauBoes) [inline src/jdk.javadoc/share/classes/jdk/javadoc/doclet/package-info.java:54]: What does the `type` attribute do? It's not mentioned in the JEP.

Comment (hns) [reply src/jdk.javadoc/share/classes/jdk/javadoc/doclet/package-info.java:54]: Should probably be `lang=java`.

**Review by hns — APPROVED** [review]
Looks good! 

Please fix wrong attribute name spotted by @FrauBoes.

**Pull request closed without merging**